### PR TITLE
Bug 1738475: Update condition reason code for invalid keyID

### PR DESCRIPTION
**Pull request by coreydaley**
A future PR will address setting the operators available false, degraded true

### CODE REVIEW

**Comment by dmage** [comment]
@coreydaley I don't see how this change can affect the Degraded condition. Have you tested it?

**Comment by coreydaley** [comment]
> @coreydaley I don't see how this change can affect the Degraded condition. Have you tested it?

I thought we had discussed on slack doing that in a separate pr that would addresss the issue of iterating over the conditions and setting the available and degraded, maybe I misunderstood.

**Comment by dmage** [comment]
I was a bit confused by `degraded true` in the description.

@adambkaplan Your comment in BZ is about additional wishes, it's not about how to fix the bug, did I get it right?

@coreydaley The referenced bug is about `panic`, when this PR gets merged, robots will move this bug to MODIFIED and QE will test it again. So we can't merge it until it fixes the bug.

File: pkg/storage/gcs/gcs.go [hunk 1]
```diff
@@ -283,7 +283,8 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 
 			if err != nil {
 				if gerr, ok := err.(*gapi.Error); ok {
-					util.UpdateCondition(cr, imageregistryv1.StorageEncrypted, operatorapi.ConditionFalse, strconv.Itoa(gerr.Code), gerr.Error())
+					util.UpdateCondition(cr, imageregistryv1.StorageEncrypted, operatorapi.ConditionFalse, "InvalidStorageConfiguration", gerr.Error())
+					return err
```
Comment (adambkaplan) [inline pkg/storage/gcs/gcs.go]: We can probably be a bit more intelligent here about the reason - ex: a 5xx error does not imply that we sent bad configuration to GCP.

**Pull request merged**